### PR TITLE
Add Peek & Pop functionality for 6/S/+

### LINKED ISSUE #29: Peek & Pop in iOS9+

### PR DESCRIPTION
**Pull request by alexthemitchell**
Closes #29

### CODE REVIEW

File: Grinnell-DB-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -600,6 +545,7 @@
 				INFOPLIST_FILE = "Grinnell-DB-iOS/Grinnell-DB-iOS-Info.plist";
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = "GrinnellAppDev.Grinnell-DB-iOSx";
```
Comment (alexthemitchell) [inline Grinnell-DB-iOS.xcodeproj/project.pbxproj]: Bundle Identifier should not change.

File: Grinnell-DB-iOS/ResultsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import UIKit
+
+
+extension ResultsViewController: UIViewControllerPreviewingDelegate {
+
+  public func previewingContext(previewingContext: UIViewControllerPreviewing, commitViewController viewControllerToCommit: UIViewController) {
+  self.navigationController!.pushViewController(viewControllerToCommit, animated: true)
+    //navigationController?.presentViewController(viewControllerToCommit, animated: true, completion: nil)
```
Comment (alexthemitchell) [inline Grinnell-DB-iOS/ResultsViewController.swift]: Remove comment

File: Grinnell-DB-iOS/Storyboard.storyboard [hunk 3]
```diff
@@ -0,0 +1,278 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (alexthemitchell) [inline Grinnell-DB-iOS/Storyboard.storyboard]: Should this file be removed?

**Comment by alexthemitchell** [comment]
Also needs tests!

**Comment by alexthemitchell** [comment]
I'm not sure how we can best test this visual effect. I'll wait until the visual rewrite to write an effective test.

**Comment by alexthemitchell** [comment]
@deweytyl Please code-review this and merge the pull request if it is satisfactory.

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 4]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
+    [tableView registerNib:[UINib nibWithNibName:@"OffCResultsCell" bundle:nil] forCellReuseIdentifier:cellIdentifier];
+  }
+
+  UITableViewCell *cell = (UITableViewCell*)[tableView dequeueReusableCellWithIdentifier:cellIdentifier];
+
+  if (!cell) {
+    cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
+  }
+
+  // Connect the cell's properties
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: I wonder if this logic belongs more in a UITableViewCell subclass

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 5]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
+  tempPerson = [searchDetails objectAtIndex:indexPath.row];
+
+  NSString *first = tempPerson.firstName;
+  NSString *last = tempPerson.lastName;
+  NSString *status = [tempPerson.attributeVals objectAtIndex:[tempPerson.attributes indexOfObject:@"Status"]];
+  NSString *username = [tempPerson.attributeVals objectAtIndex:[tempPerson.attributes indexOfObject:@"Username"]];
+
+  // Fetch the image
+  NSUInteger index = [tempPerson.attributes indexOfObject:@"picURL"];
+  if (NSNotFound != index) {
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Yoda conditional

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 6]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
+    NSString *major = [tempPerson.attributeVals objectAtIndex:[tempPerson.attributes indexOfObject:@"Major"]];
+    majorLbl.text = major;
+    classLbl.text = year;
+    statusLbl.text = status;
+  } else if ([status isEqualToString:@"Faculty / Staff"]) {
+    NSString *dept = [tempPerson.attributeVals objectAtIndex:[tempPerson.attributes indexOfObject:@"Department"]];
+
+    NSMutableArray *titleArray = [[NSMutableArray alloc] init];
+    NSUInteger index = [tempPerson.attributes indexOfObject:@"Title"];
+    while ([@"Title" isEqualToString:[tempPerson.attributes objectAtIndex:index]]) {
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Yoda conditional here too. Just a style thing

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 7]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
     if (NSNotFound != index) {
-        NSString *userImageString = [tempPerson.attributeVals objectAtIndex:index];
-        userImageView.contentMode = UIViewContentModeScaleAspectFit;
-        [userImageView sd_setImageWithURL:[NSURL URLWithString:userImageString] placeholderImage:nil];
+      NSString *title = [tempPerson.attributeVals objectAtIndex:index];
+      majorLbl.text = title;
+      classLbl.text = [tempPerson.attributeVals objectAtIndex:[tempPerson.attributes indexOfObject:@"Department"]];
     } else {
-        userImageView.image = nil;
+      majorLbl.text = @"Data unavailable off campus";
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Will this only come up when users are off campus?

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 8]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
+                          message:@"Turn on cellular data or use Wi-Fi to access the server"
                           delegate:self
                           cancelButtonTitle:@"OK"
                           otherButtonTitles:nil
                           ];
-    [error show];
+  [network show];
+}
+
+- (void)showErrorAlert {
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: This method should have a more descriptive title. What kind of error? (showNetworkErrorAlert)

Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Would it be better for this to be showGeneralErrorAlert or showUnknownErrorAlert?

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 9]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
+      NSRange startRange = [responseData rangeOfString:@"<td ></td></tr><tr><td valign = \"top\">" options:NSCaseInsensitiveSearch];
+      responseData = [self cutString:responseData fromStartToEndOfRange:startRange];
+      NSUInteger i = [selected.attributes indexOfObject:@"picURL"];
+      NSUInteger j = [selected.attributes indexOfObject:@"profileURL"];
+      NSUInteger k = [selected.attributes indexOfObject:@"Status"];
+      NSUInteger index = MIN(k, MIN(i, j));
+      if (NSNotFound == index)
+        index = selected.attributes.count;
+
+      startRange = [responseData rangeOfString:@"<TR><TD valign=\"top\" align=\"right\" ><Strong>"];
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Can this also use a case insensitive search?

File: Grinnell-DB-iOS/ResultsViewController.m (hunk 4 above)
Comment (alexthemitchell) [inline Grinnell-DB-iOS/ResultsViewController.m]: Agreed. I'm still working on the rewrite, so much of this diff is actually just whitespace transformation. Can I fix this in a future branch?

File: Grinnell-DB-iOS/ResultsViewController.m (hunk 7 above)
Comment (alexthemitchell) [inline Grinnell-DB-iOS/ResultsViewController.m]: Yes, as the majorLbl only exists on the off-campus .xib. I'm looking forward to streamlining the interface in a future Changelist.

File: Grinnell-DB-iOS/ResultsViewController.m (hunk 8 above)
Comment (alexthemitchell) [inline Grinnell-DB-iOS/ResultsViewController.m]: That's a good question, but outside the purview of this pull request

File: Grinnell-DB-iOS/ResultsViewController.m [hunk 10]
```diff
@@ -19,282 +19,281 @@ @implementation ResultsViewController
...
+                        initWithTitle:@"An Error Occurred"
+                        message:@"Please try again later"
+                        delegate:self
+                        cancelButtonTitle:@"OK"
+                        otherButtonTitles:nil
+                        ];
+  [error show];
 }
 
 - (void)parseProfilePage:(NSString *)urlString forPerson:(Person *)selected {
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: I feel like this method belongs in another class. A directory model maybe? Something to think about.

File: Grinnell-DB-iOS/ResultsViewController.swift [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import UIKit
+
+
+extension ResultsViewController: UIViewControllerPreviewingDelegate {
```
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.swift]: If this is an extension/category, i think you should use the category file-name convention. Not sure if its different in swift. But I can also understand this as a stepping stone towards rewriting the entire controller in swift.

File: Grinnell-DB-iOS/ResultsViewController.m (hunk 4 above)
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Yes please!

File: Grinnell-DB-iOS/ResultsViewController.m (hunk 8 above)
Comment (deweytyl) [inline Grinnell-DB-iOS/ResultsViewController.m]: Just giving feedback on everything i'm seeing! I know it's a bunch of changed tabs, but there's never a bad time to think about this.

**Comment by deweytyl** [comment]
LGTM

**Pull request merged**